packages/components/src/InputDate/InputDate.tsx

```diff
@@ -54,6 +58,17 @@ export function InputDate(inputProps: InputDateProps) {
                 inputProps.onBlur && inputProps.onBlur();
                 activatorProps.onBlur && activatorProps.onBlur();
                 setIsFocused(false);
+
+                if (inputProps.restoreLastValueOnBlur) {
```

Comment:
A comment here describing why we're allowing this and what issue it solves would be good. (Could even say it is experimental and may change.)

Same thing with the prop - a comment and status of experimental may help future maintainers.